### PR TITLE
OCLOMRS-308: Implement create mapping functionality

### PR DESCRIPTION
**Pull request by muhozi**
# JIRA TICKET NAME:
[Implement create mapping functionality](https://issues.openmrs.org/browse/OCLOMRS-308)

# Summary:
Implement create concept mapping functionality. Both external and internal concept mapping.

### CODE REVIEW

**Comment by Duncanian** [comment]
@muhozi , how far are you with the tests?

**Comment by muhozi** [comment]
> @muhozi , how far are you with the tests?

@Duncanian I'm finishing.

**Comment by Tittoh** [comment]
@muhozi there are a few new lines that are untested in `dictionaryConcepts.js` can you fix that?

File: src/components/dictionaryConcepts/components/ExternalMapping.jsx [hunk 1]
```diff
@@ -1,93 +1,84 @@
...
+        <small className="form-text text-danger">{errors.toDictionaryUrl}</small>
+      </FormGroup>
+      <FormGroup className="form-style">
+        Destination concept
+        <AsyncSelect
+          name="toConceptUrl"
+          placeholder="Mapping destination concept ..."
+          valueKey="id"
+          labelKey="display_name"
+          sourceUrl={conceptSearchUrl}
```
Comment (collinewait) [inline src/components/dictionaryConcepts/components/ExternalMapping.jsx]: Thanks for the effort @muhozi . You can add some spaces to contents in braces.

Comment (muhozi) [reply src/components/dictionaryConcepts/components/ExternalMapping.jsx]: @collinewait I don't think that there should not be a space according to [this](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-curly-spacing.md)

**Comment by muhozi** [comment]
> @muhozi there are a few new lines that are untested in `dictionaryConcepts.js` can you fix that?

@Tittoh I was not able to get a `moxios` option to produce a necessary exception to test untested branches.

File: src/redux/actions/concepts/dictionaryConcepts.js [hunk 2]
```diff
@@ -125,13 +128,19 @@ export const fetchDictionaryConcepts = (
   const filterByClass = getState().concepts.filteredByClass;
 
   if (filterBySource.length > 0 && filterByClass.length > 0) {
-    url = `${conceptType}/${conceptOwner}/sources/${conceptName}/concepts/?q=${query}&limit=${limit}&page=${page}&verbose=true&source=${filterBySource.join(',')}&conceptClass=${filterByClass.join(',')}`;
+    url = `${conceptType}/${conceptOwner}/collections/${conceptName}/concepts/?q=${query}&limit=${limit}&page=${page}&verbose=true&source=${filterBySource.join(
```
Comment (Duncanian) [inline src/redux/actions/concepts/dictionaryConcepts.js]: Is this change part of the ticket?

Comment (muhozi) [reply src/redux/actions/concepts/dictionaryConcepts.js]: @Duncanian No. Let me undo such unrelated changes. Thanks for this.

**Pull request closed without merging**